### PR TITLE
feat(app): return promise in exitApp

### PR DESCRIPTION
**Pull request by riderx**
document the type change in v2 of the plugin

closes https://github.com/ionic-team/capacitor-plugins/issues/750

### CODE REVIEW

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
Thanks for the PR

You also need to change `CAP_PLUGIN_METHOD(exitApp, CAPPluginReturnNone);` to `CAP_PLUGIN_METHOD(exitApp, CAPPluginReturnPromise);` in `AppPlugin.m` file.

We are not merging this yet as it's a breaking change and it's planned for when we start working in Capacitor 4 and version 2 of the plugins.

**Comment by riderx** [comment]
@jcesarmobile thanks for the report. Do we have a doc somewhere do learn oh to code and test ?

**Comment by jcesarmobile** [comment]
we test on the test-app, check its readme
https://github.com/ionic-team/capacitor-testapp#developing-capacitor

**Comment by riderx** [comment]
My issue is more about installing package and using git with Lerna, i see 0 changes do i need a specific VSCode plugin ?

**Comment by jcesarmobile** [comment]
If you run npm install on the root, lerna runs npm install on all plugins, same if you run npm run build, lerna runs it for all.
But you can also go to the plugin folder and run the same commands there, it will be faster as it will only run for that plugin instead of all.

**Comment by riderx** [comment]
@jcesarmobile it seems from the doc i have to install Lerna in global right ? https://lerna.js.org/

**Comment by jcesarmobile** [comment]
Not sure, it’s on the devDependencies, so shouldn’t be needed globally, but if it’s not working then you can give it a try.

**Comment by riderx** [comment]
@jcesarmobile yes with Global install that do way different install flow ! Now i'm able to run all commands !

**Comment by riderx** [comment]
@jcesarmobile i think that good in my side

File: app/ios/Plugin/AppPlugin.m [hunk 1]
```diff
@@ -9,5 +9,5 @@
            CAP_PLUGIN_METHOD(getLaunchUrl, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(getState, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(minimizeApp, CAPPluginReturnPromise);
-           CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnNone);
+           CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnPromise);
```
Comment (jcesarmobile) [inline app/ios/Plugin/AppPlugin.m:12]: You have to change it in `exitApp`, not in `removeAllListeners`
```suggestion
           CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnNone);
```

**Comment by riderx** [comment]
Sorry @jcesarmobile  i was sick yesterday :/ i updated

**Comment by riderx** [comment]
@jcesarmobile good for you ?

**Review by jcesarmobile — APPROVED** [review]

**Pull request merged**